### ISSUE TITLE
offline packet modification does not work

### ISSUE THREAD

**Issue by kbouna** [Montimage/5Greplay#24]
Hello, I am using following rule for offline packet modification.

```xml
root@9d2e77b65823:/opt/mmt/5greplay/rules# more registration_req_modified.xml
<beginning>
<property  value="COMPUTE" property_id="94" type_property="FORWARD"
    description="seding only malformed registration request messages" if_satisfied="#update(nas_5g.message_type, 0 )" >
    <event description="Message type: Registration request (0x41)"
           boolean_expression="( (nas_5g.message_type == 65) )"/>
</property>
</beginning>
```

the .so rule is compiled and used by the 5greplay.

```
root@9d2e77b65823:/opt/mmt/5greplay# ./5greplay info
mmt-5greplay: 5Greplay v0.0.3-f1300c5 using DPI v1.7.0.0 (a8ad3c2) is running on pid 1320
Found 1 rule.
1 - Rule id: 94
        - type            : DROP
        - events_count    : 2
        - variables_count : 1
        - variables       : nas_5g.message_type (904.2)
        - description     : seding only malformed registration request messages
        - if_satisfied    : 0x7f499caffa00
        - version         : 0.0.3 (f1300c5 - 2022-10-10 8:54:54), dpi version 1.7.0.0 (a8ad3c2)
root@9d2e77b65823:/opt/mmt/5greplay#
```

It seems that the `if_satisfied` does not happen. When using the following command to send modified packets, only the Registration request messages that exist in the pcap are replayed instead of malformed registration request messages.

Contents of the .conf file used and the respective command.
```bash
./5greplay replay -t pcap/test_proper.pcap -c 5greplay-sctp_korina.conf
```

```bash
root@9d2e77b65823:/opt/mmt/5greplay# more 5greplay-sctp_korina.conf
# option only when using DPDK to capture packets or to forwarding packets
dpdk-option = "--syslog=syslog --log-level=5 -c 0x55555555555"

#root stack of network protocol
# for Ethernet: 1
# for ieee802154: 800
# for Linux cooked capture: 624
stack-type = 1

input {
        # in case for PCAP the input mode can be ONLINE or OFFLINE, however for DPDK it's only ONLINE
        mode = ONLINE

        # input source for PCAP online mode (interface name) and for offline mode (pcap name),
        #   however for DPDK its interface port number
        #   in DPDK mode, MMT supports also multi-port inputs,
        #    - e.g., source="0,1" will tell MMT to capture packets on port 0 and 1
        #    - MMT will aggregate traffic on these 2 ports, thus 2 packets of one flow can be received on 2 different ports
        source = "pcap/test_proper.pcap"

        # maximal size of a packet
        snap-len = 65535 #
}

output {
        enable = true
        output-dir = "./" # Location where files are written:
        sample-interval = 5 #a new sample file is created each x seconds given by output.cache-period
        report-description = true # true to include rule's description into the alert reports,
             # otherwise it will be excluded (thus rules's descriptions will be an empty string in the reports)
             # Excluding rules's descriptions will reduce the size of reports.
}

engine {
        thread-nb     = 0   # the number of security threads per one probe thread , e .g . , if we have 16 probe threads and thread-nb = x ,
        # then x*16 security threads will be used .
        # If set to zero this means that the security analysis will be done by the threads of the probe .

        exclude-rules = "0-89,100-200"  # Range of rules to be excluded from the verification
        rules-mask    = ""  # Mapping of rules to the security threads:
                # Format: rules-mask = (thread-index:rule-range);
                # thread-index = a number greater than 0
                # rule-range = number greater than 0, or a range of numbers greater than 0.
                # Example: If we have thread-nb = 3 and "(1:1,2,4-6)(2:3)" ,
                # this means that:
                # thread 1 verifies rules 1 ,2 ,4 ,5 ,6;
                # thread 2 verifies only rule 3; and
                # thread 3 verifies the rest
                # Note: if we have thread-nb = 2 and "(1:1)(2:3)", then only rules 1 and 3 are verified (the others are not)

        ip-encapsulation-index = LAST   # If traffic is ip-in-ip, this option selects which IP will be analysed.
                # - FIRST: first ip in the protocol hierarchy
                # - LAST:  last ip in the protocol hierarchy
                # - i: i-th ip in ther protocol hierarchy.
                # For example, given ETH.IP.UDP.GTP.IP.TCP.VPN.IP.SSL,
                #   - FIRST, or 1, indicates IP after ETH
                #   - LAST, or any number >= 3, indicates IP after VPN
                #   - 2 indicates IP after GTP
                # NOTE: this option will be ignored in non ip-in-ip traffic

        # number of fsm instances of one rule
        max-instances = 100000
}

# A mem_pool contains several pools. Each pool stores several blocks of memory
# having the same size.
# This parameter set the maximum elements of a pool.
mempool {
        # This parameter set the Maximum bytes of a pool: 2 GBytes
        max-bytes = 2000000000
        # Max number of elements in a pool
        max-elements = 1000
        # maximum size, in bytes, of a report received from mmt-probe
        max-message-size = 3000

        # Number of reports can be stored in a ring buffer
        smp-ring-size = 1000
}


forward
{
        enable     = true
        output-nic = "lo"
        nb-copies  = 200000 #number of copies of a packet to be sent
        snap-len   = 0 #specifies the snapshot length to be set on the handle.
        promisc    = 1 #specifies whether the interface is to be put into promiscuous mode. If promisc is non-zero, promiscuous mode will be set, otherwise it will not be set.
        default    = DROP        #default action when packets are not selected/satisfied by any rule
                # either FORWARD to forward the packets or DROP to drop the packets

        #forward packets to a target using SCTP protocol: MMT will be a SCTP client,
        # - it connects to the given "sctp-host" at "sctp-port"
        # - the SCTP packets' payload will be sent to the target using this SCTP connection
        target-protocols = { SCTP}
        target-hosts     = { "10.85.245.13" }
        target-ports     = { 38412}
}
```

**Comment by nhnghia** [comment]
Unfortunately, nas_5g.message_type is not modifiable yet: https://github.com/Montimage/5Greplay/issues/22#issuecomment-1038375780

You can use this command to get the execution log of 5Greplay: `journalctl -t mmt-5greplay`
On mine, I got this message when trying your rule: `Oct 11 13:04:20 mmt mmt-5greplay[11086]: Cannot set new value 93 for att 2 of proto 904 for packet id 2579`

**Comment by nhnghia** [comment]
fixed in v0.0.4
